### ISSUE TITLE
Clarity on freeze_last_layer argument used in DinoProjectionHead

### ISSUE THREAD

**Issue by imantha-das** [lightly-ai/lightly#1586]
I have question on `freeze_last_layer` argument used in `DinoProjectionHead`. According to the docs the `freeze_last_layer` we keep the output layer fixed for an x number of epochs. I presume this is because in the Dino paper it states, "freezing the teacher network over an epoch works surprisingly well in our framework" (Under Section 3. Approach - Teacher network). However when I look at the example code for Dino in LightlySSL, they do this not for the teacher network but the student network (`self.student_head = DINOProjectionHead(
input_dim, 512, 64, 2048, freeze_last_layer=1`). So I am wondering whether this is something incorrect with the tutorial or am I misunderstanding `freeze_last_layer` argument.

**Comment by guarin** [comment]
Hi! The `freeze_last_layer` parameter corresponds to the same parameter in the original DINO repo: https://github.com/facebookresearch/dino/blob/7c446df5b9f45747937fb0d72314eb9f7b66930a/main_dino.py#L93-L95

It is used here to cancel the student projection head gradients: https://github.com/facebookresearch/dino/blob/7c446df5b9f45747937fb0d72314eb9f7b66930a/main_dino.py#L333-L334

**Comment by imantha-das** [comment]
I thought that was done (Cancel student projections heads graidents) by this line of the code in the tutorial (pytorch lightning)
```
def on_after_backward(self):
        self.student_head.cancel_last_layer_gradients(current_epoch=self.current_epoch)
```
Cause setting `freeze_last_layer = 1` only keeps the output layer fixed for 1 epoch.

**Comment by guarin** [comment]
Yes both are needed, `cancel_last_layer_gradients` checks if `current_epoch` is smaller than the `freeze_laster_layer` parameter and only then cancels the gradients, see here: https://github.com/lightly-ai/lightly/blob/08d60f7ce274a84182a07af3431e8438462a269c/lightly/models/modules/heads.py#L624-L629

**Comment by imantha-das** [comment]
Aight, thanks for the clarification !!!
So are we freezing last layer weights to get better stability. For example the initially randomized weights might effect the representations

**Comment by eharjes** [comment]
I think the `freeze_last_layer=1` argument in the DINO benchmark should be set in the student network, but it is set in the teacher network instead. This would result in, as I understand it, both the student and teacher network receiving an update in the first epoch, which is not the desired behaviour. 

https://github.com/lightly-ai/lightly/blob/ff0462787986ee0efcd816a8e994141d60e4a177/benchmarks/imagenet/resnet50/dino.py#L34

**Comment by guarin** [comment]
Hi @eharjes, thanks a lot for raising this issue, that is indeed incorrect! Should be fixed once #1596 is merged.

**Comment by guarin** [comment]
Will close this for now, please reopen if you have more questions.
